src/nvim/path.c

```diff
@@ -351,34 +356,31 @@ char_u *concat_fnames(char_u *fname1, char_u *fname2, int sep)
  * Add a path separator to a file name, unless it already ends in a path
  * separator.
  */
-void add_pathsep(char_u *p)
+void add_pathsep(char *p)
 {
-  if (*p != NUL && !after_pathsep((char *)p, (char *)p + STRLEN(p)))
-    STRCAT(p, PATHSEPSTR);
+  if (*p != NUL && !after_pathsep(p, p + strlen(p)))
+    strcat(p, PATHSEPSTR);
 }
 
-/*
- * FullName_save - Make an allocated copy of a full file name.
- * Returns NULL when fname is NULL.
- */
-char_u *
-FullName_save (
-    char_u *fname,
-    int force                      /* force expansion, even when it already looks
-                                 * like a full path name */
-)
+/// Get an allocated copy of the full path to a file.
+///
+/// @param fname is the filename to save
+/// @param force is a flag to expand `fname` even if it looks absolute
+///
+/// @return [allocated] Copy of absolute path to `fname` or NULL when
+///                     `fname` is NULL.
+char *FullName_save(char *fname, bool force)
```

Comment:
`fname` should be const. `NONNULL_RET`, `MALLOC`, and `NONNUL_ALL` could be applied here. I wouldn't normally stress this, but if we label one, we should do them all.